firedrake/__init__.py

```diff
@@ -13,6 +13,11 @@
 elif not config["options"]["honour_petsc_dir"]:  # Using our own PETSC.
     os.environ["PETSC_DIR"] = os.path.join(sys.prefix, "src", "petsc")
     os.environ["PETSC_ARCH"] = "default"
+if "PYOP2_LDFLAGS" in config["environment"]:
+    os.environ["PYOP2_LDFLAGS"] = " ".join(
+        (os.environ.get("PYOP2_LDFLAGS", default=""),
+         config["environment"]["PYOP2_LDFLAGS"])
+    )
```

Comment:
We don't use `config["environment"]` anywhere else except trivially in `firedrake-status` so this feels quite clunky and invasive for a bug fix.

Why can't we add something like`bugfix_ldflags` (like [`bugfix_cflags`](https://github.com/OP2/PyOP2/blob/master/pyop2/compilation.py#L498)) to PyOP2?

Reply:
The issue with doing this in PyOP2 is that we end up duplicating logic for the bugfix, because this fix is also needed in other parts of the Firedrake install. It therefore feels less bad to fix things in Firedrake and pass them through to PyOP2 from that single fix.

Reply:
I guess. Though that does mean that a standalone PyOP2 (which I know doesn't actually exist) on M1 is broken.